ios/browser/api/ads/brave_ads.mm

```diff
@@ -252,10 +264,20 @@ - (void)initializeWithSysInfo:(BraveAdsSysInfo*)sysInfo
   ads->Initialize(walletInfo.cppObjPtr, base::BindOnce(^(const bool success) {
                     [self periodicallyCheckForAdsResourceUpdates];
                     [self registerAdsResources];
-                    if (success) {
+                    if (success && self->adsClientNotifier != nil) {
                       self->adsClientNotifier->NotifyDidInitializeAds();
                     }
                     completion(success);
+                    if (!success) {
+                      if (self->ads != nil) {
+                        delete self->ads;
+                      }
+                      if (self->adsClientNotifier != nil) {
+                        delete self->adsClientNotifier;
+                      }
+                      self->ads = nil;
```

Comment:
See previous comment about moving inside condition, thanks

Reply:
The same pattern is used throughout the current file. So I think it is better to use it for consistency.